### PR TITLE
Full Vuex support with enhanced Counter.vue example and test spec

### PR DESCRIPTION
**Pull request by amirrustam**
This request follows up bahmutov/cypress-vue-unit-test#13 to finalize Vuex support. It works now @bahmutov 🙂

The solution required refactoring how Vue is initialized with in the app iframe. For increased thoroughness and confidence I've update the Counter example to utilize all the primary features of Vuex, and I've updated the test spec accordingly. 

vuex counter test

### CODE REVIEW

**Comment by amirrustam** [comment]
@bahmutov I'm interested to hear your initial feedback on this, because after this refactoring `vue` should really be a `peerDependency`. Since peer dependencies are not installed by default, `vue` should be installed right before the CI run. Currently, I have `vue` under dependencies because tests have to pass before commits, and I didn't want to make too many changes before getting your take.

**Comment by bahmutov** [comment]
I have not looked at it yet, but I agree - we should have Vue as a peer dependency - because the user 100% has it already. I will change this repo to do this.

File: src/index.js [hunk 1]
```diff
@@ -177,38 +144,51 @@ const mountVue = (component, optionsOrProps = {}) => () => {
...
-      return Cypress.vue
-    })
+  // display deprecation warnings
+  if (options.vue) {
+    console.warn(stripIndent`
+      [DEPRECATION]: 'vue' option has been deprecated.
+      'node_modules/vue/dis/vue' is always used.
+      Please remove it from your 'mountVue' options.`)
+  }
+
```
Comment (bahmutov) [inline src/index.js]: nice touch - showing deprecation warning

**Comment by bahmutov** [comment]
I do see error messages when running counter example 

screen shot 2018-01-31 at 8 53 31 am

But I ❤️❤️❤️ your work here, merging it in!

File: src/index.js (hunk 1 above)
Comment (amirrustam) [reply src/index.js]: I'm all about smooth transitions from version to version.

**Comment by amirrustam** [comment]
I'm on the [case](https://github.com/amirrustam/cypress-vue-unit-test/issues/1), and am investigating these warning message. But I'm glad the functionality is there now. 

I'm going to continue testing, improving, and polishing. I think this project is going to go a long way in making Cypress a one-stop-shop testing tool for Vue.js developers. Right now I'm putting it to test by testing every component in the popular [Vuetify](https://github.com/vuetifyjs/vuetify) component framework.

**Comment by bahmutov** [comment]
Perfect!

**Pull request merged**